dev/droidmate/projects/core/src/main/groovy/org/droidmate/exploration/strategy/TerminationCriterion.groovy

```diff
@@ -120,13 +120,19 @@ class TerminationCriterion implements ITerminationCriterion
   }
 
   @Override
-  void updateState()
+  void updateState(ExplorationAction performedAction)
   {
     if (timeLimited)
     {
       // Nothing to do here.
     } else
     {
+      // When shown a new runtime permission, add 2 extra available actions,
+      // one to click the runtime permission dialog "allow" button
+      // and another to re-execute the last action
+      if (performedAction.isEndorseRuntimePermission())
+        actionsLeft += 2
+
```

Comment:
delete this entire block (L130-135). I.e. allowing the permission and second click on the button shall be counted towards the actionsLeft limit.